pilot/cmd/pilot-agent/main.go

```diff
@@ -176,8 +165,9 @@ var (
 						}
 					}
 				}
-				pilotSAN = envoy.GetPilotSAN(pilotDomain, ns)
 			}
+			pilotSAN = getPilotSAN(role.DNSDomain, ns)
+			role.DNSDomain = getDNSDomain(role.DNSDomain)
```

Comment:
???

Reply:
Could you explain to us what we can change for your better understanding?

Reply:
this code is not readable. Shouldn't the sanitization occur before consuming the variable?

Reply:
We changed the order of the functions to make it more readable.
We had extracted the code to make it testable. In our original PR there was just one function for both variables. However, there was a request to have two functions. The order in which the functions are called doesn't matter.